backend/models/post_models.py

```diff
@@ -0,0 +1,24 @@
+from enum import Enum
+
+from pydantic import BaseModel
+
+
+class Post(BaseModel):
+    """Model for post metadata"""
+
+    title: str
+    content: str
+    parent_id: str | None = None
+    tags: list[str]
```

Comment:
validator for this. all tags should be alphanumeric or have slash. Use gpt or something to make a regex for it. Basically 
`<word>/<word>` or `<word>` 

Bad:
- ""
- "assignments//1"
- "assignment??/"
- "_a"
- ..